sdk/core/Azure.Core/api/Azure.Core.netstandard2.0.cs

```diff
@@ -17,6 +17,7 @@ public abstract partial class AsyncPageable<T> : System.Collections.Generic.IAsy
     }
     public static partial class AzureCoreExtensions
     {
+        public static System.Collections.Generic.Dictionary<string, object?> ToDictionaryFromJson(this System.BinaryData data) { throw null; }
```

Comment:
Based on [guidelines](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-service-return-model-collections) you shouldn't return a concrete collection/dictionary class, at least for model properties which can make mocking more difficult. Is there any reason to return a concrete type here? Is there any advantage opposed to an `IDictionary<string, object>`? That also gives us freedom to change the underlying implementation without causing a breaking change or unnecessary allocations.

Reply:
Also, is "FromJson" significant here? We don't have it for `ToObject(Async)`, or does this more align with some methods on `BinaryData` that infer the underlying data is expected to be JSON?

Reply:
Good question, I think I was conflating the `BinaryData.FromObjectAsJson` but I think you are right in this case it adds no value since technically it is "From BinaryData"